test/cache_test.lua

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env tarantool
+
+local t = require('luatest')
+local db = require('test.db')
+local ddl = require('ddl')
+local cache = require('ddl.cache')
+local helper = require('test.helper')
+
+local SPACE_NAME_IDX      = 1
+local SHARD_FUNC_NAME_IDX = 2
+local SHARD_FUNC_BODY_IDX = 3
+
+local primary_index = {
+    type = 'HASH',
+    unique = true,
+    parts = {
+        {path = 'string_nonnull', is_nullable = false, type = 'string'},
+        {path = 'unsigned_nonnull', is_nullable = false, type = 'unsigned'},
+    },
+    name = 'primary'
+}
+
+local bucket_id_idx = {
+    type = 'TREE',
+    unique = false,
+    parts = {{path = 'bucket_id', type = 'unsigned', is_nullable = false}},
+    name = 'bucket_id'
+}
+
+local func_body_first  = 'function() return 42 end'
+local func_body_second = 'function() return 24 end'
+
+local function rebuild_db(g)
+    db.drop_all()
+
+    g.space = {
+        engine = 'memtx',
+        is_local = true,
+        temporary = false,
+        format = table.deepcopy(helper.test_space_format())
+    }
+    table.insert(g.space.format, 1, {
+        name = 'bucket_id', type = 'unsigned', is_nullable = false
+    })
+
+    g.space.indexes = {
+        table.deepcopy(primary_index),
+        table.deepcopy(bucket_id_idx)
+    }
+    g.space.sharding_key = {'unsigned_nonnull', 'integer_nonnull'}
+    g.schema = {
+        spaces = {
+            space = g.space,
+        }
+    }
+end
+
+local g = t.group()
+g.before_all(db.init)
+g.before_each(function()
+    rebuild_db(g)
+end)
+
+function g.test_cache_processed_func_body()
+    g.schema.spaces.space.sharding_func = {
+        body = func_body_first
+    }
+    local ok, err = ddl.set_schema(g.schema)
+    t.assert_equals(err, nil)
+    t.assert_equals(ok, true)
+
+    local res = cache.internal.get('space')
+    t.assert(res)
+    t.assert(res.processed)
+    res = res.processed
+    t.assert(type(res) == 'function')
```

Comment:
Why do you need to create a shortcut for `res.processed` in `res` variable and cannot use it directly in assert?

Reply:
fixed

Reply:
(It seems, it is not actual anymore. Resolving the discussion.)